packages/trader/src/Modules/Trading/Components/Elements/__tests__/purchase-button.spec.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { isDesktop, isMobile } from '@deriv/shared';
+import PurchaseButton from '../purchase-button';
+
+const default_mocked_props = {
+    basis: '',
+    buy_info: { error: '' },
+    currency: '',
+    growth_rate: 0.03,
+    has_deal_cancellation: false,
+    index: 1,
+    info: {
+        has_error: false,
+        id: 'test_id',
+        message: 'test_message',
+        growth_rate: 0.03,
+        stake: '10',
+    },
+    is_accumulator: false,
+    is_disabled: false,
+    is_high_low: false,
+    is_loading: false,
+    is_multiplier: false,
+    is_proposal_empty: false,
+    is_vanilla: false,
+    is_turbos: false,
+    onClickPurchase: jest.fn(),
+    purchased_states_arr: [true, false],
+    should_fade: false,
+    setPurchaseState: jest.fn(),
+    type: 'VANILLALONGCALL',
+};
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn(() => false),
+    isDesktop: jest.fn(() => true),
+}));
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    IconTradeTypes: jest.fn(props => <p data-type={props.type}>TradeIcon</p>),
```

Comment:
@kate-deriv why p and why not data-testid ? :)

Reply:
Because `data-testid` is faceless. And when you check the presence and the value of the attribute `type` you'll already know, that there will be `type` inside. This become especially useful when there are a lot of props for testing and you can't call them all `data-testid`

Reply:
@kate-deriv since data-type is also something that doesn't exist in real component, I would recommend using data-testid just to avoid checking the attribute but instead to check if the element is present in the dom after we find it by testid: `getByTestId('').toBeInTheDocument()` :)

Reply:
But I do need to check if the passed prop is correct, not the presence of the element

Reply:
That is why I put the value in custom attribute, just to check it after

Reply:
Does this prove that the passed prop is correct? `getByTestId('call_barrier').toBeInTheDocument()` It does to me because call_barrier is present :)

Reply:
This `type` (passed as prop) is a result of function `getIconType`. It can return` empty string`, ${type.toLowerCase()}_barrier or  `type.toLowerCase()` based on some conditions. By checking type I make sure that function is working correctly
<img width="669" alt="Screenshot 2023-09-19 at 10 53 34 AM" src="https://github.com/binary-com/deriv-app/assets/121025168/61141bd2-9509-4b75-878d-ab3d2a26fb00">

Reply:
A simple check of presence of the component in the document won't help to verify the returned value from the function, as `<IconTradeTypes />` will be rendered with ANY passed `type`. And I was trying to check what is the `type` in order to make sure that  `getIconType()` worked properly with in different conditions.

Reply:
@kate-deriv i admire your great intentions!) However, the fact that `getByTestId('call_barrier').toBeInTheDocument()` is passing, and not failing, proves that the element received the correct value ('call_barrier') as its testid and the result of `getIconType()` function execution was correct. Do you agree? :)

Reply:
Agree, will replace